apps/balancer-tools/src/app/apr/(components)/MultiSelectDropdown.tsx

```diff
@@ -0,0 +1,151 @@
+"use client";
+
+import { Cross1Icon, MagnifyingGlassIcon } from "@radix-ui/react-icons";
+import Downshift, { ControllerStateAndHelpers } from "downshift";
+import React, { useCallback, useState } from "react";
+
+import { Badge } from "#/components/Badge";
+import { BaseInput } from "#/components/Input";
+
+export const MultiSelectDropdown = ({
+  items,
+  placeholderText,
+  onSelectionItemsChange,
+  initialSelectedItems,
+  ...rest
+}: {
+  items: string[];
+  placeholderText: string;
+  onSelectionItemsChange: (items: string[]) => void;
+  initialSelectedItems: string[];
+}) => {
+  const [selectedItems, setSelectedItems] =
+    useState<string[]>(initialSelectedItems);
+
+  function handleOnKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    const inputElement = event.target as HTMLInputElement;
+    if (!inputElement.value.length) {
+      if (event.key === "Backspace") {
+        const copySelectedItems = [...selectedItems];
+        copySelectedItems.pop();
+        setSelectedItems(copySelectedItems);
+        onSelectionItemsChange(copySelectedItems);
+      }
+    }
+  }
+
+  const changeHandler = useCallback(() => {
+    return (
+      selectedItem: string | null,
+      downshift: ControllerStateAndHelpers<string>,
+    ) => {
+      if (!selectedItem) return;
+      const i = selectedItems.findIndex((item) => item === selectedItem);
+      if (i === -1) setSelectedItems([...selectedItems, selectedItem]);
+      onSelectionItemsChange([...selectedItems, selectedItem]);
+      downshift.clearSelection();
+    };
+  }, [selectedItems]);
+
+  const removeSelectedItemByIndex = useCallback(
+    (i: number) => {
+      const temp = [...selectedItems];
+      temp.splice(i, 1);
+      setSelectedItems(temp);
+      onSelectionItemsChange(temp);
+    },
+    [selectedItems],
+  );
+
+  const getFilteredItems = useCallback(
+    (inputValue: string | null) => {
+      return items.filter(
+        (item) =>
+          !selectedItems.find((selected) => selected === item) &&
+          item.includes(String(inputValue)),
+      );
+    },
+    [selectedItems],
+  );
+
+  return (
+    <div className="relative">
+      <Downshift {...rest} onChange={changeHandler()}>
```

Comment:
Keyboard arrow navigation doesn't work for this Select 😞 no need to work on this now since we've already put a lot of time here, but I think any combobox should have it

Reply:
👍 Downshift should help with that, I'll take a stab at it later